src/hotspot/os_cpu/linux_riscv/vm_version_linux_riscv.cpp

```diff
@@ -74,7 +74,7 @@
 
 #define read_csr(csr)                                           \
 ({                                                              \
-        register unsigned long __v;                             \
+        unsigned long __v;                                      \
```

Comment:
Can this change be made separately?  I'd like to have the C++17 switch be as clean as possible.

Reply:
No problem!

Reply:
There are, strangely, many more register keywords in the JDK codebase than just this one, but none of them throw the same errors, only this one does

Reply:
Looks like this change has also already been made, by JDK-8319440.

All of the other non-comment uses of "register" I found in HotSpot are gcc local variable register specifications:
https://gcc.gnu.org/onlinedocs/gcc-13.2.0/gcc/Local-Register-Variables.html
So are a different thing and not affected by the deprecation/removal of the C++ "register" keyword.

Reply:
Ah, I was not aware that the asm specifications for explicit registers overrides the warning about the register keyword, thanks!

Reply:
It's not that it overrides the warning.  They are different syntactic constructs that just happen to have
a word in common.  The joys of parsing C++ ...

Reply:
I also realized that this uses a gcc statement expression currently, I wonder if this could use a lambda expression instead in another change?

Reply:
That's the standard idiom for accessing CSR, as used in libc and elsewhere. I think that avoiding divergence with such other sources is probably more important than some notion of standard purity.